plugin/core/logging.py

```diff
@@ -39,3 +39,33 @@ def exception_log(message: str, ex: Exception) -> None:
 def printf(*args: Any, prefix: str = 'LSP') -> None:
     """Print args to the console, prefixed by the plugin name."""
     print(prefix + ":", *args)
+
+
+def notify(window: Optional[sublime.Window], message: str, status: str = 'LSP: see console log…') -> None:
+    """Pick either of the 2 ways to show a user notification message:
+      - via a detailed console message and a short status message
+      - via a blocking modal dialog"""
+    from .settings import userprefs
+    if userprefs().suppress_error_dialogs:
+        if window:
+            window.status_message(status)  # print short message to statusbar
+        else:
+            sublime.status_message(status)
```

Comment:
I assume you used this to satisfy the type checker. I believe window = `None` is only a theoretical case, and if there is no (active) window, I think we don't need to log anyway. So you can just put
```
if not window:
    return
```
at the top of the function.

Reply:
done